ananse/utils.py

```diff
@@ -0,0 +1,117 @@
+import os
+import warnings
+import shutil
+import tempfile
+
+from pybedtools import BedTool
+import pysam
+
+
+def shhh_bedtool(func):
+    """
+    Decorator that silences pybedtools RuntimeWarnings such as
+    `line buffering (buffering=1) isn't supported in binary mode`
+    """
+    def wrapper(*args, **kwargs):
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", category=RuntimeWarning)
+            func(*args, **kwargs)
+    return wrapper
+
+
+@shhh_bedtool
+def bed_sort(bed):
+    """
+    Sort a bed file.
+    """
+    tmpdir = tempfile.mkdtemp()
+    tmpfile = os.path.join(tmpdir, "tmpfile")
+    try:
+        BedTool(bed).sort(output=tmpfile)
+        shutil.copy2(tmpfile, bed)
+    finally:
+        shutil.rmtree(tmpdir, ignore_errors=True)
+
+
+@shhh_bedtool
+def bed_merge(list_of_beds, merged_bed):
+    """
+    merge any number of bed files (merges overlapping regions)
+    """
+    bed = BedTool(list_of_beds[0])
+    if list_of_beds[1:]:
+        bed = bed.cat(*list_of_beds[1:])
+    bed.saveas(merged_bed)
+
+
+@shhh_bedtool
+def count_reads(bams, peakfile, bed_output):
+    """
+    Count bam reads in putative enhancer regions
+    """
+    bed = BedTool(peakfile)
+    bam_list = bams if isinstance(bams, list) else [bams]
+    bed.multi_bam_coverage(bams=bam_list, output=bed_output)
```

Comment:
Have a look at `coverage_table` from `gimmemotifs.preprocessing`. It's faster!

Reply:
this function is already no longer used. I'll add a note though :)